clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -30,7 +30,7 @@
  */
 public interface Time {
 
-    Time SYSTEM = new SystemTime();
+    Time SYSTEM = SystemTime.getInstance();
```

Comment:
Hey, is there a reason this isn't the canonical singleton instance? Why would we want two different ways to get the singleton `Time.SYSTEM` and `SystemTime.getInstance()`?

Reply:
You are right, I should deprecate `Time SYSTEM` this property, and use `SystemTime.getInstance()` to instead it.

Reply:
I would be inclined to keep the existing `Time.SYSTEM` and avoid the deprecation. SystemTime can then be package-local entirely.